clang/lib/CodeGen/CodeGenTypes.cpp

```diff
@@ -877,6 +878,11 @@ bool CodeGenTypes::isZeroInitializable(QualType T) {
   if (const MemberPointerType *MPT = T->getAs<MemberPointerType>())
     return getCXXABI().isZeroInitializable(MPT);
 
+  // HLSL Inline SPIR-V types are non-zero-initializable.
+  if (T->getAs<HLSLInlineSpirvType>()) {
+    return false;
+  }
```

Comment:
nit: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements
```suggestion
  if (T->getAs<HLSLInlineSpirvType>())
    return false;
```

Reply:
Done.